File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):

    player = dict()

    for p in participant:
        if p not in player:
            player[p] = 1
        else:
            player[p] += 1

    for c in completion:
        player[c] -= 1
    
    answer = ''
    for key in player:
        if player[key] != 0:
            answer = key
            break

    
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
  map<string, int> player;

  for(string p : participant){
    map<string, int>::iterator it = player.find(p);
    if(it == player.end())
      player[p] = 1;
    else
      player[p]++;
  }

  for(string c : completion){
    player[c]--;
  }

  string answer = "";
  map<string, int>::iterator it = player.begin();
  while(it != player.end()){
    if(it->second != 0){
      answer = it->first;
      break;
    }

    it++;
  }

    
    return answer;
}